gstd/gstd_element.c

```diff
@@ -540,39 +594,51 @@ gstd_element_fill_properties (GstdElement * self)
 
 
 static GstdReturnCode
-gstd_element_fill_signals (GstdElement * self)
+gstd_element_fill_signals_and_actions (GstdElement * self)
 {
   guint *signals;
   guint n_signals;
   GSignalQuery *query = NULL;
-  GstdObject *element_signal;
+  GstdObject *gstd_object;
+  GType type;
   guint i;
 
   g_return_val_if_fail (GSTD_IS_ELEMENT (self), GSTD_NULL_ARGUMENT);
 
-  GST_DEBUG_OBJECT (self, "Gathering \"%s\" signals", GST_OBJECT_NAME (self));
-
-  signals = g_signal_list_ids (G_OBJECT_TYPE (self->element), &n_signals);
+  GST_DEBUG_OBJECT (self, "Gathering \"%s\" signals & actions",
+      GST_OBJECT_NAME (self->element));
 
-  for (i = 0; i < n_signals; ++i) {
-    query = g_new0 (GSignalQuery, 1);
-    g_signal_query (signals[i], query);
+  for (type = G_OBJECT_TYPE (self->element); type; type = g_type_parent (type)) {
+    if (type == GST_TYPE_ELEMENT || type == GST_TYPE_OBJECT) {
+      break;
+    }
 
-    if (query->signal_flags & G_SIGNAL_ACTION) {
-      g_free (query);
+    if (type == GST_TYPE_BIN && G_OBJECT_TYPE (self->element) != GST_TYPE_BIN) {
       continue;
     }
```

Comment:
Can you explain what is going on here? Why are we filtering out some types?

Reply:
here we are looking for the signal/actions of the parent classes, like the Force-IDR that was not part of the element itself but from the parent. We break if we get an Element or Object type because those types doesn't have signals/actions and is the end of the chain. And about the bin if the parent is a bin but the element is not a bin  we don't have access to the bin actions, I took this from gst-inspect code

Reply:
By doing this, don't we lose access to the `notify::<property>` signals, for example? I believe those are `gobject` related.

* EDIT *: Indeed: https://gitlab.gnome.org/GNOME/glib/blob/master/gobject/gobject.c#L550

I think those could be very valuable for us. I don't understand the bin part.

About the bin, I still don't understand. Honestly I would leave the code install every signal in all the hierarchy chain.